lib/rake/extensiontask.rb

```diff
@@ -158,26 +157,22 @@ def define_compile_tasks(for_platform = nil, ruby_ver = RUBY_VERSION)
       directory lib_path
       directory stage_binary_dir_path
 
-      directory File.dirname(siteconf_path)
-      # Set paths for "make install" destinations
-      file siteconf_path => File.dirname(siteconf_path) do
-        File.open(siteconf_path, "w") do |siteconf|
-          siteconf.puts "require 'rbconfig'"
-          siteconf.puts "require 'mkmf'"
-          siteconf.puts "dest_path = mkintpath(#{File.expand_path(lib_path).dump})"
-          %w[sitearchdir sitelibdir].each do |dir|
-            siteconf.puts "RbConfig::MAKEFILE_CONFIG['#{dir}'] = dest_path"
-            siteconf.puts "RbConfig::CONFIG['#{dir}'] = dest_path"
-          end
-        end
-      end
-
       # copy binary from temporary location to final lib
       # tmp/extension_name/extension_name.{so,bundle} => lib/
       task "copy:#{@name}:#{platf}:#{ruby_ver}" => [lib_path, tmp_binary_path, "#{tmp_path}/Makefile"] do
         # install in lib for native platform only
         unless for_platform
-          sh "#{make} install target_prefix=", chdir: tmp_path
+          expaned_lib_path = mkintpath(File.expand_path(lib_path))
+
+          make_command = [
+            make,
+            "install",
+            "sitearchdir=#{expaned_lib_path}",
+            "sitelibdir=#{expaned_lib_path}",
+            "target_prefix=",
+          ]
+
+          sh(*make_command, chdir: tmp_path)
```

Comment:
We can't use this because `make` may have multiple shell components such as `make -j4`.